app.py

```diff
@@ -66,13 +68,109 @@ def nodes_list():
     except Exception as e:
         return jsonify({"error": str(e)})
 
+def run_fulfillment_background(network_id, requested_nodes):
+    conn = get_esi_connection(cloud=cloud_name)
+    # Run async fulfillment tasks inside this thread
+    loop = asyncio.new_event_loop()
+    asyncio.set_event_loop(loop)
+    loop.run_until_complete(fulfill_order_loop(conn, network_id, requested_nodes))
+    loop.close()
+
+async def fulfill_order_loop(conn, network_id, requested_nodes):
+    # Keep in the loop until all nodes requests fulfilled
+    while True:
+        tasks = []
+        for item in requested_nodes:
+            resource_class = item['resource_class']
+            number = item['number']
+            if number <= 0:
+                continue
+            offers_all = conn.lease.offers(resource_class=resource_class)
```

Comment:
I might recommend just getting all offers, and then filtering them by resource class in python - saves an API call.